crmsh/join_lock.py

```diff
@@ -0,0 +1,81 @@
+# Copyright (C) 2020 Xin Liang <XLiang@suse.com>
+# See COPYING for license information.
+
+
+import time
+from contextlib import contextmanager
+
+from . import bootstrap
+from . import utils
+
+
+class SSHError(Exception):
+    pass
+
+
+class JoinLock(object):
+    """
+    Class to manage lock for multiple nodes join in parallel
+    """
+
+    JOIN_LOCK_DIR = "/usr/share/crmsh/.join_lock_directory"
+    MKDIR_CMD = "mkdir {}".format(JOIN_LOCK_DIR)
+    RM_CMD = "rm -rf {}".format(JOIN_LOCK_DIR)
+    DETECT_INTERVAL = 15
+    SSH_TIMEOUT = 10
+    SSH_OPTION = "-o ConnectTimeout={} -o StrictHostKeyChecking=no".format(SSH_TIMEOUT)
+
+    def __init__(self, target_node):
+        """
+        Init function
+        """
+        self.target_node = target_node
+
+    def _run(self, cmd):
+        """
+        Run command on target node, consider specific exceptions
+        """
+        cmd_with_ssh = "ssh {} root@{} \"{}\"".format(self.SSH_OPTION, self.target_node, cmd)
+        rc, out, err = utils.get_stdout_stderr(cmd_with_ssh)
+        if rc == 255:
```

Comment:
This `255` should go in a constants to be more meaningful. What `255` does mean?

Reply:
@arbulu89 
man ssh `ssh exits with the exit status of the remote command or with 255 if an error occurred`